### PR TITLE
add new APISampling table

### PR DESCRIPTION
**Pull request by thelasttoto**

### CODE REVIEW

File: backend/pkg/database/tracesampling.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	traceSamplingTableName = "trace_sampling"
+)
+
+type TraceSampling struct {
+	gorm.Model
+
+	APIID         uint   `json:"api_id,omitempty" gorm:"column:api_id" faker:"-"`
+	TraceSourceID uint   `json:"trace_source_id,omitempty" gorm:"column:trace_source_id" faker:"-"`
+	Component     string `json:"component,omitempty" gorm:"column:component" faker:"-"`
+}
```
Comment (klyr) [inline backend/pkg/database/tracesampling.go:32]: Need to add a unique index constraint on (APIID, TraceSourceID, Component)

Comment (klyr) [reply backend/pkg/database/tracesampling.go:32]: Should we add a TraceSourceID as a foreign key ?

Comment (thelasttoto) [reply backend/pkg/database/tracesampling.go:32]: No Foreign key here, as I can have TraceSourceID==0 as valid value

Comment (thelasttoto) [reply backend/pkg/database/tracesampling.go:32]: Just seeing your PR 299... Now TraceSourceID==0 is a valid source. BUT:
- in other place (for ex apiinventory.go) TraceSourceID is not a foreign key,
- also... event_id, api_id have never foreign key. 
What missed, if any, is to delete entries from trace_sampling if a TraceSource is deleted

**Review by klyr — APPROVED** [review]
Let's go !

**Pull request merged**